src/openforms/submissions/api/validators.py

```diff
@@ -67,16 +66,17 @@ def __call__(self, form: Form, field: serializers.RelatedField):
 
 class CheckCheckboxAccepted:
     message = _("You must accept this statement.")
+    requires_context = True
 
     def __init__(self, ask_statement_field_name: str, message):
         self.ask_statement_field_name = ask_statement_field_name
         self.message = message or self.message
 
-    def __call__(self, value: bool):
-        config = GlobalConfiguration.get_solo()
-        assert isinstance(config, GlobalConfiguration)
-
-        should_statement_be_accepted = getattr(config, self.ask_statement_field_name)
+    def __call__(self, value: bool, field: serializers.BooleanField):
+        form = field.context["submission"].form
```

Comment:
With the appointments, the `submission` in the `context` can be `None` in case the that the user does not have the right permissions for the submission. Could that be a problem here?